Support for multiple input files per process

jensvannerum opened the pull request:
## References
* Fixes #9695

## Description
Introduces a new `fileParameterToBitstreamType` method in the DSpaceRunnable class.
This method is then used in `ScriptRestRepository#processFiles` to set the type of the file being processed.
Instead of hardcoding all files to "inputfile" like done before. (we still default to this if method isn't implemented by the script)

## Instructions for Reviewers
In any script, override the `fileParameterToBitstreamType` method and implement it to set the type based on the file name (or some other way). Verify multiple inputfiles can be upload on a process now as long as they differ in type.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome).
However, reviewers may request that you complete any actions in this list if you have not done so. If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [x] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR **passes Checkstyle** validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR **includes Javadoc** for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR **passes all tests and includes new/updated Unit or Integration Tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] My PR **includes details on how to test it**. I've provided clear instructions to reviewers on how to successfully test this fix or feature.
- [x] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

Commit:
116430: Support for multiple input files per process

nona-luypaert committed:
116430: Fix file name not being retrieved correctly for process files

tdonohue reviewed (changes requested):
@jensvannerum : In today's developers meeting, we reviewed this PR as a team.  Overall, the feedback was positive.  However, we had a few questions/suggestions that came up:

1. First, it would be best to add some automated tests (ITs) for the changes to ScriptRestRepository to prove the new code is working properly.
2. It was unclear to us why the multiple files **must have different types**?  Ideally, obviously, there may be scenarios where you may want to upload two files of the same type (e.g. upload two Zip files to be extracted/imported).  Is this unique type limitation something that already exists in DSpace?    We wouldn't block this PR because of this limitation, but if there's a way to avoid the limitation, it'd be nice to do so.   Alternatively, we could create a new bug ticket to document this limitation until we are able to find a better solution for uploading multiple files of the same type.

Commit:
Apply feedback to PR
- added javadoc
- added IT to prove functionality
- removed restriction which only allowed one file per type

jensvannerum commented:
@tdonohue I got some backstory about that check that disallows two files of the same type since our team originally implemented it, turns out the initial logic behind it actually isn't valid any longer and so I deleted the check.

Added a bit of javadoc to the method and added a test as requested too.
